pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +290,138 @@ def test_index_ctor_infer_periodindex(self):
         tm.assert_index_equal(rs, xp)
         assert isinstance(rs, PeriodIndex)
 
-    def test_constructor_simple_new(self):
-        idx = Index([1, 2, 3, 4, 5], name='int')
-        result = idx._simple_new(idx, 'int')
+    @pytest.mark.parametrize("vals,dtype", [
+        ([1, 2, 3, 4, 5], 'int'), ([1.1, np.nan, 2.2, 3.0], 'float'),
+        (['A', 'B', 'C', np.nan], 'obj')
+    ])
+    def test_constructor_simple_new(self, vals, dtype):
+        idx = Index(vals, name=dtype)
+        result = idx._simple_new(idx, dtype)
         tm.assert_index_equal(result, idx)
 
-        idx = Index([1.1, np.nan, 2.2, 3.0], name='float')
-        result = idx._simple_new(idx, 'float')
-        tm.assert_index_equal(result, idx)
+    @pytest.mark.parametrize("vals", [
+        [1, 2, 3], np.array([1, 2, 3]), np.array([1, 2, 3], dtype=int),
+        # below should coerce
+        [1., 2., 3.], np.array([1., 2., 3.], dtype=float)
+    ])
+    def test_constructor_dtypes_to_int64(self, vals):
```

Comment:
This is similar to a lot of the tests below and could arguably be built with parameters of "vals,dtype,klass" but I felt it was cleaner and less repetition to just break the tests by dtype